spec/features/workflow/unwithdraw_spec.rb

```diff
@@ -3,24 +3,38 @@
 RSpec.feature "Unwithdraw a document" do
   scenario do
     given_there_is_a_withdrawn_document
+    and_i_have_the_managing_editor_permission
     when_i_visit_the_summary_page
-    and_i_click_on_undo_withdraw
-    then_i_see_the_feature_is_currently_unavailable
+    and_i_click_on_undo_withdraw_and_confirm
+    then_i_see_the_document_is_now_unwithdrawn
+  end
+
+  def and_i_have_the_managing_editor_permission
+    user = User.first
+    user.update_attribute(:permissions,
+                          user.permissions + [User::MANAGING_EDITOR_PERMISSION])
   end
 
   def given_there_is_a_withdrawn_document
-    @edition = create(:edition, :withdrawn)
+    @withdrawn_edition = create(:edition, :withdrawn)
   end
 
   def when_i_visit_the_summary_page
-    visit document_path(@edition.document)
+    visit document_path(@withdrawn_edition.document)
   end
 
-  def and_i_click_on_undo_withdraw
+  def and_i_click_on_undo_withdraw_and_confirm
+    stub_publishing_api_republish(@withdrawn_edition.content_id, {})
+
     click_on "Undo withdrawal"
+    click_on(I18n.t!("documents.show.unwithdraw.confirm"))
   end
 
-  def then_i_see_the_feature_is_currently_unavailable
-    expect(page).to have_content("Sorry, this hasn't been built yet")
+  def then_i_see_the_document_is_now_unwithdrawn
+    @withdrawn_edition.reload
+
+    expect(@withdrawn_edition).to be_published
+    expect(page.body).to_not include("This #{@withdrawn_edition.document_type.label.downcase} was withdrawn")
```

Comment:
Could we make this test more robust? It feels brittle, as there's nothing in this spec that checks the text was there in the first place.

Reply:
Also, can we use I18n.t! here?

Reply:
I agree and done